mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/docker/CliShellDomainsAndUsersAdder.java

```diff
@@ -0,0 +1,69 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mpt.imapmailbox.external.james.host.docker;
+
+import java.io.IOException;
+
+import org.apache.james.mpt.imapmailbox.external.james.ProvisioningException;
+import org.apache.james.mpt.imapmailbox.external.james.host.DomainsAndUsersAdder;
+import org.apache.james.util.docker.DockerGenericContainer;
+import org.testcontainers.containers.Container;
+import org.testcontainers.shaded.org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.ImmutableList;
+
+public class CliShellDomainsAndUsersAdder implements DomainsAndUsersAdder {
+
+    private final DockerGenericContainer container;
+
+    public CliShellDomainsAndUsersAdder(DockerGenericContainer container) {
+        this.container = container;
+    }
+
+    @Override
+    public void addDomain(String domain) throws Exception {
+        String cli = getCliPath();
+        Container.ExecResult execResult = container.exec(cli, "-h", "127.0.0.1", "-p", "9999", "adddomain", domain);
```

Comment:
Everything is duplicated between the Jar User and domain adder & the cli one. Except the CLI prefix

Can we consider simplifying a bit all of this?